packages/fdc3-get-agent/src/ui/AbstractUIComponent.ts

```diff
@@ -66,27 +86,40 @@ export abstract class AbstractUIComponent implements Connectable {
 
   async messagePortReady(port: MessagePort) {
     // tells the iframe it can start posting
-    const handshake: BrowserTypes.Fdc3UserInterfaceHandshake = {
+    const message: Fdc3UserInterfaceHandshake = {
       type: 'Fdc3UserInterfaceHandshake',
       payload: {
         fdc3Version: FDC3_VERSION,
       },
     };
-    port.postMessage(handshake);
+    port.postMessage(message);
   }
 
   private awaitHello(): Promise<MessagePort> {
-    return new Promise((resolve, _reject) => {
+    return new Promise((resolve /*, _reject*/) => {
       const ml = (e: MessageEvent) => {
-        if (e.source == this.iframe?.contentWindow && e.data.type == BrowserTypes.FDC3_USER_INTERFACE_HELLO_TYPE) {
-          const helloData = e.data as BrowserTypes.Fdc3UserInterfaceHello;
-          if (helloData.payload.initialCSS) {
-            this.themeContainer(helloData.payload.initialCSS);
+        if (e.source == this.iframe?.contentWindow) {
+          if (isFdc3UserInterfaceHello(e.data)) {
+            const helloData = e.data;
+            if (helloData.payload.initialCSS) {
+              this.themeContainer(helloData.payload.initialCSS);
+            }
+            const port = e.ports[0];
+            port.start();
+            globalThis.window.removeEventListener('message', ml);
+            resolve(port);
+          } else {
+            Logger.debug('AbstractUIComponent: ignored UI Message from UI iframe while awaiting hello: ', e.data);
           }
-          const port = e.ports[0];
-          port.start();
-          globalThis.window.removeEventListener('message', ml);
-          resolve(port);
+        } else {
+          Logger.debug(
```

Comment:
Just a log statement on an unexpected case. We could drop the else clause and nothing will break, but it may leave someone with a difficult debug one day - went in while I was have troubl eiwth message routing in the tests.